scipy/stats/_stats_py.py

```diff
@@ -4583,6 +4618,14 @@ def pearsonr(x, y, *, alternative='two-sided'):
         * 'greater':  the correlation is positive (greater than zero)
 
         .. versionadded:: 1.9.0
+    method : PermutationMethod, optional
+        Defines the method used to compute the p-value. If `method` is an
+        instance of `PermutationMethod`, the p-value is computed using
```

Comment:
Defining and testing a `MonteCarloMethod` would also make sense here. Let's save that for a follow-up after multi-sample MC testing is supported by `monte_carlo_test`.